demisto_sdk/commands/download/downloader.py

```diff
@@ -160,271 +141,316 @@ def __init__(
+
+        Returns:
+            int: Exit code. 1 if failed, 0 if succeeded
         """
-        if not self.verify_flags():
-            return 1
-        if not self.download_system_item and not self.fetch_custom_content():
-            return 1
-        if self.download_system_item and not self.fetch_system_content():
-            return 1
-        if self.handle_list_files_flag():
-            return 0
-        self.handle_init_flag()
-        self.handle_all_custom_content_flag()
-        self.handle_regex_flag()
-        if not self.verify_output_pack_is_pack():
-            return 1
+        try:
+            if not (self.verify_flags() and self.verify_output_path()):
+                return 1
 
-        self.build_pack_content()
-        self.build_custom_content() if not self.download_system_item else self.build_system_content()
-        self.update_pack_hierarchy()
-        self.merge_into_pack()
-        self.log_files_downloaded()
-        self.log_files_not_downloaded()
+            if self.init:
+                self.initialize_output_path()
 
-        for entry in self.files_not_downloaded:
-            file, reason = entry
-            if reason != FILE_EXIST_REASON:
-                return 1
+            if self.download_system_items and not self.list_files:
+                downloaded_content_objects = self.fetch_system_content()
+
+            else:
+                all_custom_content_data = self.download_custom_content()
+                all_custom_content_objects = self.parse_custom_content_data(custom_content_data=all_custom_content_data)
+
+                # If we're in list-files mode, print the list of all available custom content and exit
+                if self.list_files:
+                    logger.info(f"list-files (-lf) mode detected. Listing available custom content files "
+                                f"({len(all_custom_content_objects)}):")
+                    table_str = self.create_custom_content_table(custom_content_objects=all_custom_content_objects)
+
+                    logger.info(table_str)
+                    return 0
+
+                uuid_mapping = self.create_uuid_to_name_mapping(custom_content_objects=all_custom_content_objects)
+
+                # Filter custom content so that we'll process only downloaded content
+                downloaded_content_objects = self.filter_custom_content(
+                    custom_content_objects=all_custom_content_objects)
+
+                # Replace UUID IDs with names in filtered content (only content we download)
+                changed_uuids_count = 0
+                for _, file_object in downloaded_content_objects.items():
+                    if self.replace_uuid_ids(custom_content_object=file_object, uuid_mapping=uuid_mapping):
+                        changed_uuids_count += 1
+
+                if changed_uuids_count > 0:
+                    logger.debug(f"Replaced UUID IDs in {changed_uuids_count} custom content items.")
```

Comment:
let's log their names/ids as well, sorted.

Reply:
You mean log what was replaced?
If so, there is a log within `self.replace_uuid_ids`:
```python
logger.debug(f"Replacing UUID '{uuid}' with '{uuid_mapping[uuid]}' in "
             f"'{custom_content_object['name']}'")
```
(line 432)

It's in debug level since there can be a few hundreds of these lines.